junit/core/src/main/java/org/jboss/arquillian/junit/Arquillian.java

```diff
@@ -81,6 +81,12 @@ public Arquillian(Class<?> testClass) throws InitializationError {
 
     @Override
     public void run(final RunNotifier notifier) {
+        if (State.hasAnyArquillianRule(this.getTestClass())) {
+            throw new RuntimeException(String.format("TestClass: %s contains Arquillian runner and Arquillian Rule."
+                + " Arquillian doesn't support @RunWith(Arquillian.class) and ArquillianTestClass or "
+                + "ArquillianTest to use at the same time.", this.getTestClass().getName()));
```

Comment:
Maybe also inform the user that he should decide whether to runner or rules - ideally with the link to the docs.

Reply:
Also - if we enforce the user to always define both rules we prevent from strange bugs being opened. In other words - if not both rules specified maybe we should fail the test?